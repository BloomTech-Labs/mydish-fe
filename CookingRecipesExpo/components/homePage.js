import React from 'react';
import Search from "./Search.js";

import {
  SafeAreaView,
} from 'react-native';


const HomePage = (props) => {

  React.useEffect(() => {
    // console.log('props in HomePage', props.navigation);
  },[])

  
  return (
      <SafeAreaView>
        <Search/>
      </SafeAreaView>
  );
};


export default HomePage;