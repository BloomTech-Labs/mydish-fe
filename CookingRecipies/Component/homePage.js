import React from 'react';
import styles from '../styles/styles.js';
import Search from "./Search.js";

import {
  SafeAreaView,
  StyleSheet,
  ScrollView,
  View,
  Text,
  StatusBar,
} from 'react-native';


const HomePage = () => {
  return (

      <SafeAreaView
      // style={{
      //   flex: 1, 
      //   justifyContent: "center", 
      //   alignItems: "center"
      //  }}
       >
        {/* <StatusBar barStyle="dark-content" /> */}
        <Search/>
        {/* <RecipeList/> */}
        <Text style={styles.red}>Hello home page</Text>
      </SafeAreaView>

  );
};

HomePage.navigationOptions = {
    title: 'Home',
    };

export default HomePage;