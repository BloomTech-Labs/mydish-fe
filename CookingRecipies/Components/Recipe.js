import React, {useState, useEffect}from 'react';
import styles from '../styles/recipe-styles';
import {View,Text,ScrollView, Image, TouchableOpacity} from 'react-native';
import { withNavigation } from 'react-navigation'
// import {Icon} from 'react-native-elements';
// import Icon from "react-native-vector-icons/FontAwesome";
// import ipad from '../assets/ipadrecipe.jpg';
// const ipad = require('../assets/ipadrecipe.jpg');
import styled from 'styled-components';
import clearHeart from '../assets/clear-heart.png';
import solidHeart from '../assets/solid-heart.png';


var Cereal = "https://image.shutterstock.com/z/stock-photo-cornflakes-with-milk-in-the-white-bowl-322906217.jpg"
var HomeFries = "https://image.shutterstock.com/z/stock-photo-fried-potatoes-147539354.jpg"
var Eggplant = "https://image.shutterstock.com/image-photo/grilled-eggplants-seasoned-olive-oil-260nw-87708241.jpg"
var ScrambledEggs = "https://image.shutterstock.com/image-photo/mexican-food-recipes-revoltillo-de-600w-752977636.jpg"
let fakeImages=["Cereal","Home Fries","Eggplant","Scrambled Eggs"]

const getRandomInt = (min, max) =>
  Math.floor(Math.random() * (max - min + 1)) + min

const Recipe = (props) => {
    const [num, setNum]= useState(1)
    console.log("jeight", props.height)
    let [like, setLike] = React.useState(false);


    const im = ()=>{
        if(props.recipe.img==null){
            return(
                <Image 
                
                source={{uri : Cereal}}
                style={{width: 160, height: props.height, borderRadius: 20, paddingRight: 20 }}
                resieMode="contain"
                
            />
            )
        }else{
            return(
                <Image 
                
                source={{uri : props.recipe.img}}
                style={{width: 160, height: props.height, borderRadius: 20, paddingRight: 20 }}
                resieMode="contain"
                
            />
            )
        }
    }

    
    //   useEffect(() =>{
    //     if (num !== 0){
    //         setNum()
    //         setNum(0)
    //         console.log("0", num)
    //     } 
    //     else{
    //         setNum()
    //         setNum(1)
    //         console.log("1 ", num)
    //     }
        
    //   }, [])
     
    // let [likeCount, setLikeCount] = React.useState(props.recipe.likes)  //get likes from recipe handed down via props from the database.

    // useEffect(() =>{
    //     for(i in fakeImages){
    //         console.log("i", fakeImages[i], props.recipe.title)
    //         if(fakeImages[i] == props.recipe.title=="Cereal"){
    //             console.log("hi")
    //             setImg(Cereal)
    //             console.log(img)
    //         }
    //         if(fakeImages[i] == props.recipe.title=="Home Fries"){
    //             setImg(HomeFries)
    //         }
    //     }
    //     console.log('props in recipe', img);
    // },[]);
        const RecipeCard = styled.View`
        justifyContent: flex-start;
        width: 130;
        marginLeft: 20;
        marginRight: 13;
        marginBottom: 110;
        `;

    const UserCard = styled.View`
      flexDirection : row;
      justifyContent : flex-start;
    `;

    const Like = styled.View`
        flexDirection: row;
        position: absolute;
        left : 10;
        top: 5;
        zIndex : 1;
    `;

    // const RecipeImage = styled.View`
    //     marginRight: 30;
    //     paddingRight: 10;
    // `;

    const likeIt = () => {
        console.log('like pressed');
        setLike(!like);
    }


    return (
        <>
            {/* <View style={{flex: 1, minWidth: 160, alignItems: 'center'}}> */}
            <RecipeCard style={{height: props.height}}>
               <TouchableOpacity  
               onPress={()  =>  props.navigation.navigate('IndividualR', {paramsID: props.recipe.id})}
               >
                {/* <Like onStartShouldSetResponder={likeIt}>
                    <Image source={like ? solidHeart : clearHeart } style={{width: 20, height: 20}}/>
                    <Text style={{color : 'white', fontWeight: 'bold'}}> 3k</Text>
                </Like> */}
                {im()}
                <Text style={styles.text}>{props.recipe.title}</Text>
                <UserCard>
                    <Image source={{uri : "https://fakeimg.pl/50x50/?text=user"}}
                        style={{width: 50, height: 50 }}/>
                    <View style={styles.usercardTxt}>
                        <Text style={styles.username}>{props.recipe.username}</Text>
                        <Text style={styles.prep}>{props.recipe.minutes} min.</Text>
                    </View>
                </UserCard>
                 </TouchableOpacity>
            </RecipeCard>
        </>
    )
}

export default withNavigation(Recipe);