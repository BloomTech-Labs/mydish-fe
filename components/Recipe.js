import React from 'react';
import styles from '../styles/recipe-styles';
import { View, Text, Image, TouchableOpacity } from 'react-native';
import { withNavigation } from 'react-navigation';
import { savedPlaceholder } from '../constants/imagePlaceholders';
import { maxUsername } from '../constants/maxLength';
//Analytics
import { Analytics, Event } from 'expo-analytics';
const analytics = new Analytics('UA-160806654-1');

const Recipe = ({ navigation, recipe, parent }) => {
  const totalCookTime = (recipe.prep_time || 0) + (recipe.cook_time || 0);

  const handlePress = () => {
    navigation.navigate('IndividualR', {
      recipeID: recipe.id,
    });
    analytics
      .event(new Event('Recipe', 'User taps on recipe card'))
      .then(() => console.log('Recipe tapped'))
      .catch((e) => console.log(e.message));
  };

  return (
    <TouchableOpacity
      style={
        parent === 'Cookbook'
          ? styles.cookbookContainer
          : styles.recipeContainer
      }
      onPress={handlePress}
    >
      <Image
        source={recipe.img ? { uri: recipe.img } : savedPlaceholder}
        style={{
          width: parent === 'Cookbook' ? '38%' : '100%',
          height: parent === 'Cookbook' ? 75 : 130,
          borderRadius: 5,
        }}
      />

      <Text style={parent === 'Cookbook' ? styles.cookbookText : styles.text}>
        {recipe.title}
      </Text>

      <View
        style={
          parent === 'Cookbook' ? styles.cookbookPrepView : styles.prepView
        }
      >
        {!(parent === 'Cookbook') && (
          <>
            <Text style={styles.prepAndUsername}>
              {recipe.owner.username.length > maxUsername
                ? `${recipe.owner.username.slice(0, maxUsername)}...`
                : recipe.owner.username}
            </Text>
            <Text style={styles.prepAndUsername}>{totalCookTime} min.</Text>
          </>
        )}
      </View>
    </TouchableOpacity>
  );
};

export default withNavigation(Recipe);
