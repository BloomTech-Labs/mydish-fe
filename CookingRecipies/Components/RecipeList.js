import React from 'react';
import styles from '../styles/styles';
import Recipe from './Recipe';
import {AppRegistry, View} from 'react-native';
// import {name as appName} from '../app.json';
import App from '../App';

const RecipeList = () => {

    return (
        <>
        <View/>
            <Recipe/>
         <View/>
         </>
    )  
}

// AppRegistry.registerComponent(appName, () => App );

export default RecipeList;

