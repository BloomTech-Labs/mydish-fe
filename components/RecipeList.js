import React from 'react';
import { useSelector } from 'react-redux';
import Recipe from './Recipe';
import { View, FlatList } from 'react-native';
import FancySpinner from './FancySpinner';

const RecipeList = ({ parent, folder }) => {
  const isLoading = useSelector((store) => store.cookbook.isLoading);

  // If the RecipeList is being rendered from the cookbook,
  //     grab the props that the cookbook is passing down.
  // If the RecipeList ISN'T coming from the cookbook, then we
  //     useSelector to get all of the recipes.
  const recipeList =
    parent === 'cookbook'
      ? folder
      : useSelector((store) => store.allRecipes.recipeList);

  if (isLoading) return <FancySpinner />;

  return (
    <View style={{ flexDirection: 'row', flexWrap: 'wrap' }}>
      {recipeList.length !== 0 && (
        <FlatList
          contentContainerStyle={{ paddingBottom: 150 }}
          data={recipeList}
          numColumns={2}
          keyExtractor={(item) => item.id.toString()}
          renderItem={({ item }) => <Recipe key={item.id} recipe={item} />}
        />
      )}
    </View>
  );
};

export default RecipeList;
