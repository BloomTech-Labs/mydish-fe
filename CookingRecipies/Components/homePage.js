import React from 'react';
import styles from '../styles/styles.js';
import Search from "./Search.js";
import RecipeList from './RecipeList.js'

import {
  SafeAreaView,
  StyleSheet,
  ScrollView,
  View,
  Text,
  StatusBar,
} from 'react-native';


const HomePage = () => {
  return (
      <SafeAreaView>

        <Search/>
      
      </SafeAreaView>


  );
};

HomePage.navigationOptions = {
    title: 'Home',
    };

export default HomePage;