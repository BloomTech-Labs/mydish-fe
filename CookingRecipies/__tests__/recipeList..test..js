import React from 'react';
import RecipeList from '../Components/RecipeList';
import Recipe from '../Components/Recipe';
import {render, fireEvent} from 'react-native-testing-library';
import renderer from 'react-test-renderer';
import styled from 'styled-components/native';
import homePage from '../Components/homePage';

  test('testing react-native-testing-library', () => {
    const RL = render(<RecipeList/>);
    // console.log("recipe list rendered by react native testing library render() : ", RL)
    const RLSnap = renderer.create(<RecipeList/>).toJSON();
    expect(RLSnap).toMatchSnapshot();
  })

  test('testing homePage', () => {
    const HP = render(<homePage/>);
    console.log('homepage', HP);
    HPSnap = renderer.create(<homePage/>).toJSON();
    expect(HPSnap).toMatchSnapshot();
    const Text = HP.getByText('Hello home page');
    console.log(Text);
    expect(text).toBe('Hello home page');
  })






  



  // test('renders correctly', () => {
  //   const homePageSnap = renderer.create(<homePage/>).toJSON();
  //   expect(homePageSnap).toMatchSnapshot();
  // });


  // it('recipelist', () => {
  //   const recipeList = renderer.create(<RecipeList/>).toJSON();
  //   expect(recipeList).toMatchSnapshot();
  // })






