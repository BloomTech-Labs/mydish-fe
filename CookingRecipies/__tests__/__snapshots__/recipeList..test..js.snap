// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing homePage 1`] = `<homePage />`;

exports[`testing react-native-testing-library 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "flex-start",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "flex-start",
              "marginBottom": 10,
              "marginLeft": 10,
              "marginRight": 10,
              "minWidth": 150,
            },
          ]
        }
      >
        <View
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
                "left": 10,
                "position": "absolute",
                "top": 5,
                "zIndex": 1,
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../assets/clear-heart.png",
              }
            }
            style={
              Object {
                "height": 20,
                "width": 20,
              }
            }
          />
          <Text
            style={
              Object {
                "color": "white",
                "fontWeight": "bold",
              }
            }
          >
             3k
          </Text>
        </View>
        <Image
          resieMode="contain"
          source={
            Object {
              "uri": "https://fakeimg.pl/250x100/?text=recipe",
            }
          }
          style={
            Object {
              "borderRadius": 20,
              "height": 300,
              "paddingRight": 20,
              "width": 190,
            }
          }
        />
        <Text
          style={
            Object {
              "fontSize": 20,
              "fontWeight": "bold",
              "marginBottom": 10,
              "marginTop": 10,
              "textAlign": "left",
              "width": 150,
            }
          }
        >
          Sample Recipe
        </Text>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "justifyContent": "flex-start",
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "uri": "https://fakeimg.pl/50x50/?text=user",
              }
            }
            style={
              Object {
                "height": 50,
                "width": 50,
              }
            }
          />
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginLeft": 5,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 15,
                }
              }
            >
              Username
            </Text>
            <Text
              style={
                Object {
                  "fontSize": 15,
                }
              }
            />
            <Text
              style={
                Object {
                  "alignSelf": "flex-end",
                  "marginStart": -40,
                }
              }
            >
              Min: Prep Time
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "flex-start",
              "marginBottom": 10,
              "marginLeft": 10,
              "marginRight": 10,
              "minWidth": 150,
            },
          ]
        }
      >
        <View
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
                "left": 10,
                "position": "absolute",
                "top": 5,
                "zIndex": 1,
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../assets/clear-heart.png",
              }
            }
            style={
              Object {
                "height": 20,
                "width": 20,
              }
            }
          />
          <Text
            style={
              Object {
                "color": "white",
                "fontWeight": "bold",
              }
            }
          >
             3k
          </Text>
        </View>
        <Image
          resieMode="contain"
          source={
            Object {
              "uri": "https://fakeimg.pl/250x100/?text=recipe",
            }
          }
          style={
            Object {
              "borderRadius": 20,
              "height": 130,
              "paddingRight": 20,
              "width": 190,
            }
          }
        />
        <Text
          style={
            Object {
              "fontSize": 20,
              "fontWeight": "bold",
              "marginBottom": 10,
              "marginTop": 10,
              "textAlign": "left",
              "width": 150,
            }
          }
        >
          Sample Recipe
        </Text>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "justifyContent": "flex-start",
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "uri": "https://fakeimg.pl/50x50/?text=user",
              }
            }
            style={
              Object {
                "height": 50,
                "width": 50,
              }
            }
          />
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginLeft": 5,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 15,
                }
              }
            >
              Username
            </Text>
            <Text
              style={
                Object {
                  "fontSize": 15,
                }
              }
            />
            <Text
              style={
                Object {
                  "alignSelf": "flex-end",
                  "marginStart": -40,
                }
              }
            >
              Min: Prep Time
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "flex-start",
              "marginBottom": 10,
              "marginLeft": 10,
              "marginRight": 10,
              "minWidth": 150,
            },
          ]
        }
      >
        <View
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
                "left": 10,
                "position": "absolute",
                "top": 5,
                "zIndex": 1,
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../assets/clear-heart.png",
              }
            }
            style={
              Object {
                "height": 20,
                "width": 20,
              }
            }
          />
          <Text
            style={
              Object {
                "color": "white",
                "fontWeight": "bold",
              }
            }
          >
             3k
          </Text>
        </View>
        <Image
          resieMode="contain"
          source={
            Object {
              "uri": "https://fakeimg.pl/250x100/?text=recipe",
            }
          }
          style={
            Object {
              "borderRadius": 20,
              "height": 300,
              "paddingRight": 20,
              "width": 190,
            }
          }
        />
        <Text
          style={
            Object {
              "fontSize": 20,
              "fontWeight": "bold",
              "marginBottom": 10,
              "marginTop": 10,
              "textAlign": "left",
              "width": 150,
            }
          }
        >
          Sample Recipe
        </Text>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "justifyContent": "flex-start",
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "uri": "https://fakeimg.pl/50x50/?text=user",
              }
            }
            style={
              Object {
                "height": 50,
                "width": 50,
              }
            }
          />
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginLeft": 5,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 15,
                }
              }
            >
              Username
            </Text>
            <Text
              style={
                Object {
                  "fontSize": 15,
                }
              }
            />
            <Text
              style={
                Object {
                  "alignSelf": "flex-end",
                  "marginStart": -40,
                }
              }
            >
              Min: Prep Time
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "flex-start",
              "marginBottom": 10,
              "marginLeft": 10,
              "marginRight": 10,
              "minWidth": 150,
            },
          ]
        }
      >
        <View
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
                "left": 10,
                "position": "absolute",
                "top": 5,
                "zIndex": 1,
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../assets/clear-heart.png",
              }
            }
            style={
              Object {
                "height": 20,
                "width": 20,
              }
            }
          />
          <Text
            style={
              Object {
                "color": "white",
                "fontWeight": "bold",
              }
            }
          >
             3k
          </Text>
        </View>
        <Image
          resieMode="contain"
          source={
            Object {
              "uri": "https://fakeimg.pl/250x100/?text=recipe",
            }
          }
          style={
            Object {
              "borderRadius": 20,
              "height": 130,
              "paddingRight": 20,
              "width": 190,
            }
          }
        />
        <Text
          style={
            Object {
              "fontSize": 20,
              "fontWeight": "bold",
              "marginBottom": 10,
              "marginTop": 10,
              "textAlign": "left",
              "width": 150,
            }
          }
        >
          Sample Recipe
        </Text>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "justifyContent": "flex-start",
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "uri": "https://fakeimg.pl/50x50/?text=user",
              }
            }
            style={
              Object {
                "height": 50,
                "width": 50,
              }
            }
          />
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginLeft": 5,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 15,
                }
              }
            >
              Username
            </Text>
            <Text
              style={
                Object {
                  "fontSize": 15,
                }
              }
            />
            <Text
              style={
                Object {
                  "alignSelf": "flex-end",
                  "marginStart": -40,
                }
              }
            >
              Min: Prep Time
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "flex-start",
              "marginBottom": 10,
              "marginLeft": 10,
              "marginRight": 10,
              "minWidth": 150,
            },
          ]
        }
      >
        <View
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
                "left": 10,
                "position": "absolute",
                "top": 5,
                "zIndex": 1,
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../assets/clear-heart.png",
              }
            }
            style={
              Object {
                "height": 20,
                "width": 20,
              }
            }
          />
          <Text
            style={
              Object {
                "color": "white",
                "fontWeight": "bold",
              }
            }
          >
             3k
          </Text>
        </View>
        <Image
          resieMode="contain"
          source={
            Object {
              "uri": "https://fakeimg.pl/250x100/?text=recipe",
            }
          }
          style={
            Object {
              "borderRadius": 20,
              "height": 300,
              "paddingRight": 20,
              "width": 190,
            }
          }
        />
        <Text
          style={
            Object {
              "fontSize": 20,
              "fontWeight": "bold",
              "marginBottom": 10,
              "marginTop": 10,
              "textAlign": "left",
              "width": 150,
            }
          }
        >
          Sample Recipe
        </Text>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "justifyContent": "flex-start",
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "uri": "https://fakeimg.pl/50x50/?text=user",
              }
            }
            style={
              Object {
                "height": 50,
                "width": 50,
              }
            }
          />
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginLeft": 5,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 15,
                }
              }
            >
              Username
            </Text>
            <Text
              style={
                Object {
                  "fontSize": 15,
                }
              }
            />
            <Text
              style={
                Object {
                  "alignSelf": "flex-end",
                  "marginStart": -40,
                }
              }
            >
              Min: Prep Time
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "flex-start",
              "marginBottom": 10,
              "marginLeft": 10,
              "marginRight": 10,
              "minWidth": 150,
            },
          ]
        }
      >
        <View
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
                "left": 10,
                "position": "absolute",
                "top": 5,
                "zIndex": 1,
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../assets/clear-heart.png",
              }
            }
            style={
              Object {
                "height": 20,
                "width": 20,
              }
            }
          />
          <Text
            style={
              Object {
                "color": "white",
                "fontWeight": "bold",
              }
            }
          >
             3k
          </Text>
        </View>
        <Image
          resieMode="contain"
          source={
            Object {
              "uri": "https://fakeimg.pl/250x100/?text=recipe",
            }
          }
          style={
            Object {
              "borderRadius": 20,
              "height": 130,
              "paddingRight": 20,
              "width": 190,
            }
          }
        />
        <Text
          style={
            Object {
              "fontSize": 20,
              "fontWeight": "bold",
              "marginBottom": 10,
              "marginTop": 10,
              "textAlign": "left",
              "width": 150,
            }
          }
        >
          Sample Recipe
        </Text>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "justifyContent": "flex-start",
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "uri": "https://fakeimg.pl/50x50/?text=user",
              }
            }
            style={
              Object {
                "height": 50,
                "width": 50,
              }
            }
          />
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginLeft": 5,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 15,
                }
              }
            >
              Username
            </Text>
            <Text
              style={
                Object {
                  "fontSize": 15,
                }
              }
            />
            <Text
              style={
                Object {
                  "alignSelf": "flex-end",
                  "marginStart": -40,
                }
              }
            >
              Min: Prep Time
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "flex-start",
              "marginBottom": 10,
              "marginLeft": 10,
              "marginRight": 10,
              "minWidth": 150,
            },
          ]
        }
      >
        <View
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
                "left": 10,
                "position": "absolute",
                "top": 5,
                "zIndex": 1,
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../assets/clear-heart.png",
              }
            }
            style={
              Object {
                "height": 20,
                "width": 20,
              }
            }
          />
          <Text
            style={
              Object {
                "color": "white",
                "fontWeight": "bold",
              }
            }
          >
             3k
          </Text>
        </View>
        <Image
          resieMode="contain"
          source={
            Object {
              "uri": "https://fakeimg.pl/250x100/?text=recipe",
            }
          }
          style={
            Object {
              "borderRadius": 20,
              "height": 300,
              "paddingRight": 20,
              "width": 190,
            }
          }
        />
        <Text
          style={
            Object {
              "fontSize": 20,
              "fontWeight": "bold",
              "marginBottom": 10,
              "marginTop": 10,
              "textAlign": "left",
              "width": 150,
            }
          }
        >
          Sample Recipe
        </Text>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "justifyContent": "flex-start",
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "uri": "https://fakeimg.pl/50x50/?text=user",
              }
            }
            style={
              Object {
                "height": 50,
                "width": 50,
              }
            }
          />
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginLeft": 5,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 15,
                }
              }
            >
              Username
            </Text>
            <Text
              style={
                Object {
                  "fontSize": 15,
                }
              }
            />
            <Text
              style={
                Object {
                  "alignSelf": "flex-end",
                  "marginStart": -40,
                }
              }
            >
              Min: Prep Time
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "flex-start",
              "marginBottom": 10,
              "marginLeft": 10,
              "marginRight": 10,
              "minWidth": 150,
            },
          ]
        }
      >
        <View
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
                "left": 10,
                "position": "absolute",
                "top": 5,
                "zIndex": 1,
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../assets/clear-heart.png",
              }
            }
            style={
              Object {
                "height": 20,
                "width": 20,
              }
            }
          />
          <Text
            style={
              Object {
                "color": "white",
                "fontWeight": "bold",
              }
            }
          >
             3k
          </Text>
        </View>
        <Image
          resieMode="contain"
          source={
            Object {
              "uri": "https://fakeimg.pl/250x100/?text=recipe",
            }
          }
          style={
            Object {
              "borderRadius": 20,
              "height": 130,
              "paddingRight": 20,
              "width": 190,
            }
          }
        />
        <Text
          style={
            Object {
              "fontSize": 20,
              "fontWeight": "bold",
              "marginBottom": 10,
              "marginTop": 10,
              "textAlign": "left",
              "width": 150,
            }
          }
        >
          Sample Recipe
        </Text>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "justifyContent": "flex-start",
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "uri": "https://fakeimg.pl/50x50/?text=user",
              }
            }
            style={
              Object {
                "height": 50,
                "width": 50,
              }
            }
          />
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginLeft": 5,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 15,
                }
              }
            >
              Username
            </Text>
            <Text
              style={
                Object {
                  "fontSize": 15,
                }
              }
            />
            <Text
              style={
                Object {
                  "alignSelf": "flex-end",
                  "marginStart": -40,
                }
              }
            >
              Min: Prep Time
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "flex-start",
              "marginBottom": 10,
              "marginLeft": 10,
              "marginRight": 10,
              "minWidth": 150,
            },
          ]
        }
      >
        <View
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
                "left": 10,
                "position": "absolute",
                "top": 5,
                "zIndex": 1,
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../assets/clear-heart.png",
              }
            }
            style={
              Object {
                "height": 20,
                "width": 20,
              }
            }
          />
          <Text
            style={
              Object {
                "color": "white",
                "fontWeight": "bold",
              }
            }
          >
             3k
          </Text>
        </View>
        <Image
          resieMode="contain"
          source={
            Object {
              "uri": "https://fakeimg.pl/250x100/?text=recipe",
            }
          }
          style={
            Object {
              "borderRadius": 20,
              "height": 300,
              "paddingRight": 20,
              "width": 190,
            }
          }
        />
        <Text
          style={
            Object {
              "fontSize": 20,
              "fontWeight": "bold",
              "marginBottom": 10,
              "marginTop": 10,
              "textAlign": "left",
              "width": 150,
            }
          }
        >
          Sample Recipe
        </Text>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "justifyContent": "flex-start",
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "uri": "https://fakeimg.pl/50x50/?text=user",
              }
            }
            style={
              Object {
                "height": 50,
                "width": 50,
              }
            }
          />
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginLeft": 5,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 15,
                }
              }
            >
              Username
            </Text>
            <Text
              style={
                Object {
                  "fontSize": 15,
                }
              }
            />
            <Text
              style={
                Object {
                  "alignSelf": "flex-end",
                  "marginStart": -40,
                }
              }
            >
              Min: Prep Time
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
